speechbrain/processing/features.py

```diff
@@ -993,6 +996,169 @@ def forward(self, x):
         return cw_x
 
 
+def gaussian_statistics(x: torch.Tensor, dim: Union[int, tuple, None] = None):
+    """
+    Compute first- and second-order moments of data, and return them as the
+    count, mean, and variance of a vector over one or more dimensions.
+
+    Arguments
+    ---------
+    x: torch.Tensor
+        The tensor to compute the statistics over
+    dim: int | tuple | None
+        The dimension or dimensions that the statistics should be computed over.
+        The other dimensions are retained in the output.
+        If None, then scalar-valued statistics will be returned.
+
+    Returns
+    -------
+    count
+        The number of sub-vectors or sub-tensors that the statistics were
+        computed over.
+    mean
+        The mean.
+    variance
+        The variance.
+    """
+
+    if dim is None:
+        number = math.prod(x.shape)
+    elif isinstance(dim, int):
+        number = x.shape[dim]
+    else:
+        assert isinstance(dim, tuple)
+        if dim == ():
+            return 1, x, torch.zeros_like(x)
```

Comment:
Mainly I was thinking of this edge case, I'm not too concerned about someone passing an empty tuple here, and if they did I'm guessing it would break in predictable ways. But honestly this is only two lines, so, who cares?

Reply:
Meaning we can probably just leave it as-is.